src/io/filesystem.ts

```diff
@@ -344,91 +182,76 @@ export async function readFileContent(file: FileFileEntry): Promise<EditorConten
       }
     }
     default: {
-      const textContent = import.meta.env.VITE_IS_WEB
-        ? await readTextFileFromWeb(currentProjectId, file.path)
-        : await readTextFile(systemPath);
-
+      const textContent = await readProjectTextFile(currentProjectId, file.path);
       return { type: 'text', textContent };
     }
   }
 }
 
 export async function deleteFile(relativePath: string): Promise<boolean> {
   try {
-    if (import.meta.env.VITE_IS_WEB) {
-      await deleteFileFromWeb(currentProjectId, relativePath);
-      return true;
-    } else {
-      const systemPath = await getSystemPath(relativePath);
-      await removeFile(systemPath);
-      return true;
-    }
+    await deleteProjectFile(currentProjectId, relativePath);
+    return true;
   } catch (err) {
     console.error('Error', err);
     return false;
   }
 }
 
-export async function renameFile(relativePath: string, newName: string): RenameFileResult {
+export async function renameFile(relativePath: string, newRelativePath: string): RenameFileResult {
+  if (relativePath.split(getSeparator()).length > 1) {
+    console.warn(`Only root files can be renamed. Found ${relativePath}`);
+    return { success: false };
+  }
   try {
-    if (import.meta.env.VITE_IS_WEB) {
-      const newRelativePath = getRefStudioPath(newName);
-      if (await existsFileFromWeb(currentProjectId, newRelativePath)) {
-        console.warn(`Another file with the same name already exists: ${newRelativePath}`);
-        return { success: false };
-      }
-      await renameFileFromWeb(currentProjectId, relativePath, newRelativePath);
-      return { success: true, newPath: newRelativePath };
-    } else {
-      const systemPath = await getSystemPath(relativePath);
-      const newSystemPath = await join(getParentFolder(systemPath), newName);
-      if (await exists(newSystemPath)) {
-        console.warn(`Another file with the same name already exists: ${newSystemPath}`);
-        return { success: false };
-      }
-      await tauriRenameFile(systemPath, newSystemPath);
-      const newRelativePath = getRefStudioPath(newSystemPath);
-      return { success: true, newPath: newRelativePath };
+    if (await existsProjectFile(currentProjectId, newRelativePath)) {
+      console.warn(`Another file with the same name already exists: ${newRelativePath}`);
```

Comment:
shouldn't this check go in `renameProjectFile`?

Reply:
that's correct